integration-tests/sync-tests/sync-verifier.spec.ts

```diff
@@ -0,0 +1,94 @@
+import chai, { expect } from 'chai'
+import { Wallet, BigNumber, providers } from 'ethers'
+import { injectL2Context } from '@eth-optimism/core-utils'
+
+import { sleep, l2Provider } from '../test/shared/utils'
+import { OptimismEnv } from '../test/shared/env'
+import { DockerComposeNetwork } from '../test/shared/docker-compose'
+
+describe('Syncing a verifier', () => {
+  let env: OptimismEnv
+  let wallet: Wallet
+  let verifier: DockerComposeNetwork
+
+  const provider = injectL2Context(l2Provider)
+
+  /* Helper functions */
+
+  const waitForBatchSubmission = async (totalElementsBefore: BigNumber) => {
+    // Wait for batch submission to happen by watching the CTC
+    let totalElementsAfter = (await env.ctc.getTotalElements()) as BigNumber
+    while (totalElementsBefore.eq(totalElementsAfter)) {
+      await sleep(500)
+      totalElementsAfter = (await env.ctc.getTotalElements()) as BigNumber
+    }
+    return totalElementsAfter
+  }
+
+  const startAndSyncVerifier = async (sequencerBlockNumber: number) => {
+    // Bring up new verifier
+    verifier = new DockerComposeNetwork(['verifier'])
+    await verifier.up({ commandOptions: ['--scale', 'verifier=1'] })
+
+    // Wait for verifier to be looping
+    let logs = await verifier.logs()
+    while (!logs.out.includes('Starting Sequencer Loop')) {
+      await sleep(500)
+      logs = await verifier.logs()
+    }
+
+    const verifierProvider = injectL2Context(
+      new providers.JsonRpcProvider('http://localhost:8547')
+    )
+
+    // Wait until verifier has caught up to the sequencer
+    let latestVerifierBlock = (await verifierProvider.getBlock('latest')) as any
```

Comment:
Is there a type in `core-utils` that represents the L2 block? At one point I think there was one in the batch submitter

Reply:
There is not, we could add it in a separate PR as it warrants a bigger refactor probably.